app/View/Label.php

```diff
@@ -49,7 +48,24 @@ public function render(callable $callback = null)
                 ->with('count', $this->data->get('count'));
         }
 
-        $template->validate();
+        try {
+            $template = LabelModel::find($settings->label2_template);
+
+            if ($template === null) {
+                throw new \UnexpectedValueException('Template is null.');
+            }
+
+            $template->validate();
+        } catch (\UnexpectedValueException $e) {
+
+            \Log::error('Validation failed: ' . $e->getMessage());
+
+        } catch (\Throwable $e) {
+
+            \Log::error('An unexpected error occurred: ' . $e->getMessage());
+
+        }
```

Comment:
We'll need to do something after catching the exception instead of only logging since `$template` actually being null will throw an exception on line 71.

To test this you can change line 52 to `$template = LabelModel::find('asdf');`.

Can we easily do a `return redirect()` with an error message for the user?

Reply:
added a null check and redirect on the render and the save methods